### PR TITLE
Fix retry defaults and error messages

### PR DESCRIPTION
**Pull request by emilgoldsmith**
closes https://github.com/tgymnich/fork-sync/issues/129
closes https://github.com/tgymnich/fork-sync/issues/112

Maybe some more it closes too?

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -80,13 +81,13 @@ function run() {
             }
         }
         catch (error) {
-            if (error.request.request.retryCount) {
+            if ((_d = (_c = error === null || error === void 0 ? void 0 : error.request) === null || _c === void 0 ? void 0 : _c.request) === null || _d === void 0 ? void 0 : _d.retryCount) {
                 console.log(`request failed after ${error.request.request.retryCount} retries with a delay of ${error.request.request.retryAfter}`);
             }
-            if (!!error.errors && !!error.errors[0] && !!error.errors[0].message && error.errors[0].message.startsWith('No commits between')) {
+            if ((_k = (_j = (_h = ((_e = error === null || error === void 0 ? void 0 : error.errors) !== null && _e !== void 0 ? _e : (_g = (_f = error === null || error === void 0 ? void 0 : error.response) === null || _f === void 0 ? void 0 : _f.data) === null || _g === void 0 ? void 0 : _g.errors)) === null || _h === void 0 ? void 0 : _h[0]) === null || _j === void 0 ? void 0 : _j.message) === null || _k === void 0 ? void 0 : _k.startsWith('No commits between')) {
```
Comment (Nickronomicon) [inline lib/main.js:87]: fwiw, you can make this a lot more readable with something like:
```js
// `val` if `val` exists else `false`.
function _e (val) { return val !== null && val !== void 0 ? val : false; };

val errors = _e(_e(error).errors) || _e(_e(_e(_e(error).response).data).errors);

if(_e(_e(_e(_e(errors).[0]).message).startsWith('No commits between'))) {
    console.log('No commits between ' + context.repo.owner + ':' + base + ' and ' + owner + ':' + head);
}
```

Comment (emilgoldsmith) [reply lib/main.js:87]: This is transpiled code

Comment (Nickronomicon) [reply lib/main.js:87]: Ha! I'm a dummy, I didn't even see the .ts right below it.

Comment (emilgoldsmith) [reply lib/main.js:87]: No problem haha :)

**Comment by emilgoldsmith** [comment]
Yes you can use any branch as a GitHub action. I don’t think this branch is in a working state though because tgymnich does some stuff on separate branches to include the node modules when he deploys, but my master branch should work for you I think, did for me anyway

File: src/main.ts [hunk 2]
```diff
@@ -49,14 +49,14 @@ async function run() {
...
   } catch (error: any) {
-    if (error.request.request.retryCount) {
+    if (error?.request?.request?.retryCount) {
       console.log(
         `request failed after ${error.request.request.retryCount} retries with a delay of ${error.request.request.retryAfter}`
       );
     }
-    if (!!error.errors && !!error.errors[0] && !!error.errors[0].message && error.errors[0].message.startsWith('No commits between')) {
+    if ((error?.errors ?? error?.response?.data?.errors)?.[0]?.message?.startsWith('No commits between')) {
       console.log('No commits between ' + context.repo.owner + ':' + base + ' and ' + owner + ':' + head);
```
Comment (parkermo) [inline src/main.ts:58]: Can we pass the error message here like in line 60? `console.log(error.errors[0].message);`

**Review by parkermo — APPROVED** [review]

**Pull request merged**